client/src/main/java/se/fortnox/reactivewizard/client/ReactorRxClientProvider.java

```diff
@@ -58,8 +62,13 @@ private HttpClient buildClient(InetSocketAddress socketAddress) {
                 })
             )
             .port(config.getPort())
-            .doOnRequest((httpClientRequest, connection) -> healthRecorder.logStatus(connectionProvider, true))
-            .doOnError((httpClientRequest, throwable) -> healthRecorder.logStatus(connectionProvider, false), (httpClientResponse, throwable) -> { })
+            .doOnRequest((httpClientRequest, connection) -> {
+                errorCount.set(0);
+                healthRecorder.logStatus(connectionProvider, true);
+            })
+            .doOnError((httpClientRequest, throwable) -> {
+                healthRecorder.logStatus(connectionProvider, errorCount.incrementAndGet() < config.getNumberOfConnectionFailuresAllowed());
```

Comment:
Won't this be false when there are 10 errors even if 10 are configured as allowed?

Your test also says "First nine should not cause the client to be unhealthy", but that conflicts with the configuration saying "Number of connection failures allowed". 
If I configure to allow 10 errors then 11 should be considered unhealthy, right?

Reply:
Correct

Reply:
I fix